design/ceph/object/store.md

```diff
@@ -351,3 +351,218 @@ spec:
+                ]
+            }
+        ],
+        "default_placement": "default-placement",
+         ...
+    }
+    ```
+5. update json to:
+    ```json
+    {
+        "placement_targets": [
+            {
+                "name": "hot",
+                "tags": [],
+                "storage_classes": [
+                    "STANDARD"
+                ]
+            },
+            {
+                "name": "cold",
+                "tags": [],
+                "storage_classes": [
+                    "STANDARD","ARCHIVE"
+                ]
+            }
+        ],
+        "default_placement": "cold",
+         ...
+    }
+    ```
+6. update with `radosgw-admin zonegroup set --rgw-zonegroup=<CephObjectStore name> --infile zonegroup.json` and `radosgw-admin period update --commit`
+7. get default zone: `radosgw-admin zone get --rgw-zone=<CephObjectStore name>`
+    ```json
+    {
+        "placement_pools": [
+            {
+                "key": "default-placement",
+                "val": {
+                    "index_pool": "<CephObjectStore name>.rgw.buckets.index",
+                    "storage_classes": {
+                        "STANDARD": {
+                            "data_pool": "<CephObjectStore name>.rgw.buckets.data"
+                        }
+                    },
+                    "data_extra_pool": "<CephObjectStore name>.rgw.buckets.non-ec",
+                    "index_type": 0,
+                    "inline_data": true
+                }
+            }
+        ],
+        ...
+    }
+    ```
+8. update json to 
+    ```json
+    {
+        "placement_pools": [
+            {
+                "key": "hot",
+                "val": {
+                    "index_pool": "hot.rgw.buckets.index",
+                    "storage_classes": {
+                        "STANDARD": {
+                            "data_pool": "hot.rgw.buckets.data"
+                        }
+                    },
+                    "data_extra_pool": "hot.rgw.buckets.non-ec",
+                    "index_type": 0,
+                    "inline_data": true
+                }
+            },
+            {
+                "key": "cold",
+                "val": {
+                    "index_pool": "cold.rgw.buckets.index",
+                    "storage_classes": {
+                        "STANDARD": {
+                            "data_pool": "cold.rgw.buckets.data"
+                        },
+                        "ARCHIVE": {
+                            "data_pool": "glacier.rgw.buckets.data"
+                        }
+                    },
+                    "data_extra_pool": "cold.rgw.buckets.non-ec",
+                    "index_type": 0,
+                    "inline_data": true
+                }
+            }
+        ],
+        ...
+    }
+    ```
+9. update with `radosgw-admin zone set --rgw-zone=<CephObjectStore name> --infile zone.json` and `radosgw-admin period update --commit`
+10. restart CephObjectStore pods
```

Comment:
The way we restart pods is by updating some property on the pod spec, so we need a property to update. A hash of the targetPlacement settings might be simplest. Or perhaps an incrementing number every time the period is updated. But the hash seems simpler.

Reply:
@cbodley do RGWs need to be restarted after these zone updates, or can they pick up the updates while online?

Reply:
rgw restarts aren't necessary. the `radosgw-admin period update --commit` command will cause all running radosgws to reload the realm/zone configuration at runtime

Reply:
@cbodley There is note in Ceph docs mentioning that restart is required if multisite is not configured. Is it still relevant?

> If you have not done any previous [Multisite Configuration](https://docs.ceph.com/en/latest/radosgw/multisite), a default zone and zonegroup are created for you, and changes to the zone/zonegroup will not take effect until the Ceph Object Gateways are restarted. If you have created a realm for multisite, the zone/zonegroup changes will take effect once the changes are committed with radosgw-admin period update --commit.

https://docs.ceph.com/en/latest/radosgw/placement/#zonegroup-zone-configuration

Reply:
my understanding is that rook always creates a realm, even for single-site ObjectStores. the presence of a realm is enough to use 'period commit' and not need restarts